app/services/quarterly_external_income_export.rb

```diff
@@ -0,0 +1,58 @@
+class QuarterlyExternalIncomeExport
+  HEADERS = [
+    "RODA identifier",
+    "Delivery partner identifier",
+    "Delivery partner organisation",
+    "Title",
+    "Level",
+    "Providing organisation",
+    "ODA",
+  ]
+
+  def initialize(organisation, source_fund_code)
+    @organisation = organisation
+    @source_fund_code = source_fund_code
+    activity_ids = Activity.where(organisation_id: @organisation.id, source_fund_code: @source_fund_code).pluck(:id)
+    @activity_external_income = ExternalIncome.includes(:activity, :organisation).where(activity_id: activity_ids).order(:activity_id, :financial_year, :financial_quarter)
+    @financial_quarters = @activity_external_income.map(&:own_financial_quarter).uniq
+  end
+
+  def headers
+    return HEADERS if @activity_external_income.empty?
+
+    HEADERS + financial_quarter_range.map(&:to_s)
+  end
+
+  def rows
+    @activity_external_income.map do |record|
+      [
+        record.activity.roda_identifier,
+        record.activity.delivery_partner_identifier,
+        @organisation.name,
+        record.activity.title,
+        I18n.t("table.body.activity.level.#{record.activity.level}"),
+        record.organisation.name,
+        I18n.t("table.body.external_income.oda_funding.#{record.oda_funding?}"),
+      ] + fq_row(record)
+    end
+  end
+
+  private
+
+  def fq_row(activity_external_income)
+    # return [] if @activity_external_income.empty?
```

Comment:
Did you mean to leave this in?

Reply:
No 😬 It was part of debugging, but it should be uncommented.